core/src/main/java/hivemall/regression/GeneralRegressionUDTF.java

```diff
@@ -0,0 +1,131 @@
+import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
+
+/**
+ * A general regression class with replaceable optimization functions.
+ */
+@Description(name = "train_regression",
+        value = "_FUNC_(list<string|int|bigint> features, double label [, const string options])"
+                + " - Returns a relation consists of <string|int|bigint feature, float weight>",
+        extended = "Build a prediction model by a generic regressor")
+@Since(version = "0.5-rc.1")
+public final class GeneralRegressionUDTF extends RegressionBaseUDTF {
+
+    @Nonnull
+    private final Map<String, String> optimizerOptions;
+    private Optimizer optimizer;
+    private LossFunction lossFunction;
+
+    public GeneralRegressionUDTF() {
+        super(true); // This enables new model interfaces
+        this.optimizerOptions = OptimizerOptions.create();
+    }
+
+    @Override
+    public StructObjectInspector initialize(ObjectInspector[] argOIs) throws UDFArgumentException {
+        if (argOIs.length != 2 && argOIs.length != 3) {
+            throw new UDFArgumentException(this.getClass().getSimpleName()
+                    + " takes 2 or 3 arguments: List<Text|Int|BitInt> features, float target "
+                    + "[, constant string options]");
+        }
+
+        StructObjectInspector outputOI = super.initialize(argOIs);
+
+        if (lossFunction.forBinaryClassification()) {
+            throw new UDFArgumentException("The loss function `" + lossFunction + "` is not for regression");
+        }
+        if (is_mini_batch) {
+            throw new UDFArgumentException("_FUNC_ does not currently support `-mini_batch` option");
+        }
+
+        try {
+            this.optimizer = createOptimizer(optimizerOptions);
+        } catch (Throwable e) {
+            throw new UDFArgumentException(e.getMessage());
+        }
+
+        return outputOI;
+    }
+
+    @Override
+    protected Options getOptions() {
+        Options opts = super.getOptions();
+        opts.addOption("loss", "loss_function", true,
+                "Loss function [default: SquaredLoss, QuantileLoss, EpsilonInsensitiveLoss]");
+        OptimizerOptions.setup(opts);
+        return opts;
+    }
+
+    @Override
+    protected CommandLine processOptions(ObjectInspector[] argOIs) throws UDFArgumentException {
+        CommandLine cl = super.processOptions(argOIs);
+        try {
+            if (cl.hasOption("loss_function")) {
+                this.lossFunction = LossFunctions.getLossFunction(cl.getOptionValue("loss_function"));
+            } else {
+                this.lossFunction = LossFunctions.getLossFunction("SquaredLoss");
+            }
+        } catch (Throwable e) {
+            throw new UDFArgumentException(e.getMessage());
+        }
+        OptimizerOptions.propcessOptions(cl, optimizerOptions);
+        return cl;
+    }
+
+    @Override
+    protected final void checkTargetValue(final float target) throws UDFArgumentException {
```

Comment:
@maropu This is a regressor which simply predicts real values. Why did you create this method? Values only in [0,1] are allowed...?